src/coreclr/jit/gentree.cpp

```diff
@@ -26915,6 +27024,12 @@ bool GenTreeHWIntrinsic::OperIsEmbRoundingEnabled() const
         case NI_AVX512DQ_ConvertToVector512Double:
         case NI_AVX512DQ_ConvertToVector512Int64:
         case NI_AVX512DQ_ConvertToVector512UInt64:
+        case NI_AVX10v1_ConvertToInt32:
```

Comment:
Are we missing `NI_Avx10v1_X64_ConvertToInt64` and `NI_Avx10v1_X64_ConvertToUInt64`?

Reply:
I think we missed this in the initial design of ```Avx10v1```. I see there are set of APIs in ```Avx512F_X64``` which are supported by ```Avx10v1```. 


```cpp
//  AVX512F.X64 Intrinsics
HARDWARE_INTRINSIC(AVX512F_X64,     ConvertScalarToVector128Double,             16,             -1,     false,  {INS_invalid,           INS_invalid,            INS_invalid,            INS_invalid,            INS_invalid,            INS_invalid,            INS_cvtsi2sd64,         INS_vcvtusi2sd64,       INS_invalid,            INS_invalid},           HW_Category_SIMDScalar,             HW_Flag_BaseTypeFromSecondArg|HW_Flag_CopyUpperBits|HW_Flag_SpecialCodeGen|HW_Flag_EmbRoundingCompatible)
HARDWARE_INTRINSIC(AVX512F_X64,     ConvertScalarToVector128Single,             16,             -1,     false,  {INS_invalid,           INS_invalid,            INS_invalid,            INS_invalid,            INS_invalid,            INS_invalid,            INS_cvtsi2ss64,         INS_vcvtusi2ss64,       INS_invalid,            INS_invalid},           HW_Category_SIMDScalar,             HW_Flag_BaseTypeFromSecondArg|HW_Flag_CopyUpperBits|HW_Flag_SpecialCodeGen|HW_Flag_EmbRoundingCompatible)
HARDWARE_INTRINSIC(AVX512F_X64,     ConvertToInt64,                             16,             -1,     false,  {INS_invalid,           INS_invalid,            INS_invalid,            INS_invalid,            INS_invalid,            INS_invalid,            INS_invalid,            INS_invalid,            INS_cvtss2si,           INS_cvtsd2si},          HW_Category_SIMDScalar,             HW_Flag_BaseTypeFromFirstArg|HW_Flag_SpecialCodeGen|HW_Flag_EmbRoundingCompatible)
HARDWARE_INTRINSIC(AVX512F_X64,     ConvertToUInt64,                            16,             -1,     false,  {INS_invalid,           INS_invalid,            INS_invalid,            INS_invalid,            INS_invalid,            INS_invalid,            INS_invalid,            INS_invalid,            INS_vcvtss2usi,         INS_vcvtsd2usi},        HW_Category_SIMDScalar,             HW_Flag_BaseTypeFromFirstArg|HW_Flag_SpecialCodeGen|HW_Flag_EmbRoundingCompatible)
HARDWARE_INTRINSIC(AVX512F_X64,     ConvertToUInt64WithTruncation,              16,              1,      true,  {INS_invalid,           INS_invalid,            INS_invalid,            INS_invalid,            INS_invalid,            INS_invalid,            INS_invalid,            INS_invalid,            INS_vcvttss2usi64,      INS_vcvttsd2usi64},     HW_Category_SIMDScalar,             HW_Flag_BaseTypeFromFirstArg|HW_Flag_SpecialCodeGen)
```

Do we want to duplicate them in the ```Avx10v1``` API interface i.e. add all the APIs in ```Avx10v1``` or inherit ```Avx512F_X64``` to ```Avx10v1_X64```.

```c#
class Avx10v1: Avx2 {
    class X64: Avx512F.X64 {
    
    }
}
```

Both of them seems fine to me. inheriting them would avoid duplication but we are duplicating much part of API surface so what would be better here?

Reply:
Inheriting the base X64 class is better. It should be the same base class as the parent inherits from (so if Avx10v1.V512 inherits Avx512BW, then Avx10v1.V512.X64 should inherit Avx512BW.X64

Reply:
Makes sense. I will add those and other missing x64 APIs in ```Avx10v1``` and ```Avx10v1.V512```